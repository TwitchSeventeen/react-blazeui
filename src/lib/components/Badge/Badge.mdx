---
name: Badge
---

import { Playground, PropsTable } from 'docz'
import Badge from './index';

# Badge

They're useful when you want to add additional info. Like a version number of a project or for status messages for individual items.

<Playground>
    <Badge>34 unread messages</Badge>
</Playground>

## Colours
Badges come as default, brand, info, warning, success and error

To use, add the **type** prop:

<Playground>
    <Badge>Default</Badge>
    <br />
    <Badge type="brand">Brand</Badge>
    <br />
    <Badge type="info">Info</Badge>
    <br />
    <Badge type="warning">Warning</Badge>
    <br />
    <Badge type="success">Success</Badge>
    <br />
    <Badge type="error">Error</Badge>
</Playground>

## Rounded
Make your badges rounded with the **rounded** prop.

<Playground>
    <Badge rounded>Default</Badge>
    <br />
    <Badge type="brand" rounded>Brand</Badge>
    <br />
    <Badge type="info" rounded>Info</Badge>
    <br />
    <Badge type="warning" rounded>Warning</Badge>
    <br />
    <Badge type="success" rounded>Success</Badge>
    <br />
    <Badge type="error" rounded>Error</Badge>
</Playground>

## Ghost
Remove your badges bodies with the **ghost** prop.

<Playground>
    <Badge rounded ghost>Default</Badge>
    <br />
    <Badge type="brand" rounded ghost>Brand</Badge>
    <br />
    <Badge type="info" rounded ghost>Info</Badge>
    <br />
    <Badge type="warning" rounded ghost>Warning</Badge>
    <br />
    <Badge type="success" rounded ghost>Success</Badge>
    <br />
    <Badge type="error" rounded ghost>Error</Badge>
</Playground>

## Properties
<PropsTable of={Badge} />