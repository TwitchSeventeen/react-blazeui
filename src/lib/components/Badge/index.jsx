import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Badge(props) {
  const {
    children, type, rounded, ghost, dataName, ...other
  } = props;
  const classes = classNames(
    'c-badge',
    { [`c-badge--${type}`]: type },
    { 'c-badge--rounded': rounded },
    { 'c-badge--ghost': ghost },
  );

  return (
    <span
      {...other}
      className={classes}
      data-name={dataName}
    >
      { children }
    </span>
  );
}

Badge.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * A badge with rounded corners
   */
  rounded: PropTypes.bool,
  /**
   * A badge without a filled body
   */
  ghost: PropTypes.bool,
  /**
   * The type of button
   */
  type: PropTypes.oneOf([
    'brand',
    'info',
    'warning',
    'success',
    'error',
  ]),
  /**
   * Optional data-name attribute
   * (use for testing/selection)
   */
  dataName: PropTypes.string,
};

Badge.defaultProps = {
  children: null,
  rounded: false,
  ghost: false,
  type: null,
  dataName: 'badge',
};

export default Badge;
