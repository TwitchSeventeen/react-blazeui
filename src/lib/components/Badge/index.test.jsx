import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Badge from './index';

describe('Badge', () => {
  afterEach(cleanup);

  it('should be defined', () => {
    expect(Badge).toBeDefined();
  });

  it('renders correctly', () => {
    const { asFragment } = render(
      <Badge>lorem</Badge>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with a type', () => {
    const { asFragment } = render(
      <Badge type="success">lorem</Badge>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with rounded corners', () => {
    const { asFragment } = render(
      <Badge type="error" rounded>lorem</Badge>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly a ghost button', () => {
    const { asFragment } = render(
      <Badge ghost>lorem</Badge>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
