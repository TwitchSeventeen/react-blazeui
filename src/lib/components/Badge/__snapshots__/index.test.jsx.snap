// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders correctly 1`] = `
<DocumentFragment>
  <span
    class="c-badge"
    data-name="badge"
  >
    lorem
  </span>
</DocumentFragment>
`;

exports[`Badge renders correctly a ghost button 1`] = `
<DocumentFragment>
  <span
    class="c-badge c-badge--ghost"
    data-name="badge"
  >
    lorem
  </span>
</DocumentFragment>
`;

exports[`Badge renders correctly with a type 1`] = `
<DocumentFragment>
  <span
    class="c-badge c-badge--success"
    data-name="badge"
  >
    lorem
  </span>
</DocumentFragment>
`;

exports[`Badge renders correctly with rounded corners 1`] = `
<DocumentFragment>
  <span
    class="c-badge c-badge--error c-badge--rounded"
    data-name="badge"
  >
    lorem
  </span>
</DocumentFragment>
`;
